From f390a643a65d73ec09ba7c234dac7ec8ac99c87b Mon Sep 17 00:00:00 2001
From: vijaykumar@bravegnu.org <vijaykumar@bravegnu.org>
Date: Wed, 29 Oct 2008 08:58:32 +0530
Subject: [PATCH] Staging: poch: Block size bug fix

Block size is to be expressed in no. of 64 bit transfers. But the code
specifies the block size in bytes. Fix this issue.

Signed-off-by: Vijay Kumar <vijaykumar@bravegnu.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/poch/poch.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/poch/poch.c b/drivers/staging/poch/poch.c
index 0e113f9..de3f8c6 100644
--- a/drivers/staging/poch/poch.c
+++ b/drivers/staging/poch/poch.c
@@ -432,7 +432,10 @@ static void channel_dma_init(struct channel_info *channel)
 	}
 
 	printk(KERN_WARNING "block_size, group_size, group_count\n");
-	iowrite32(channel->block_size, fpga + block_size_reg);
+	/*
+	 * Block size is represented in no. of 64 bit transfers.
+	 */
+	iowrite32(channel->block_size / 8, fpga + block_size_reg);
 	iowrite32(channel->group_size / channel->block_size,
 		  fpga + block_count_reg);
 	iowrite32(channel->group_count, fpga + group_count_reg);
-- 
1.5.6.5

